examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -14,58 +14,54 @@
 // limitations under the License.
 // </copyright>
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net.Http;
-using Examples.AspNetCore.Models;
+namespace Examples.AspNetCore.Controllers;
+using System.Diagnostics;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.Extensions.Logging;
 
-namespace Examples.AspNetCore.Controllers
+[ApiController]
+[Route("[controller]")]
+public class WeatherForecastController : ControllerBase
 {
-    [ApiController]
-    [Route("[controller]")]
-    public class WeatherForecastController : ControllerBase
+    public static readonly string ActivitySourceName = "Examples.AspNetCore";
+    private static readonly ActivitySource ActivitySource = new(ActivitySourceName, "1.0.0");
+
+    private static readonly string[] Summaries = new[]
     {
-        private static readonly string[] Summaries = new[]
-        {
-            "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
-        };
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
+    };
 
-        private static readonly HttpClient HttpClient = new HttpClient();
+    private readonly ILogger<WeatherForecastController> logger;
+    private readonly ExampleMeter exampleMeter;
 
-        private readonly ILogger<WeatherForecastController> logger;
+    public WeatherForecastController(ILogger<WeatherForecastController> logger, ExampleMeter exampleMeter)
+    {
+        this.logger = logger;
+        this.exampleMeter = exampleMeter;
+    }
 
-        public WeatherForecastController(ILogger<WeatherForecastController> logger)
+    [HttpGet(Name = "GetWeatherForecast")]
+    public IEnumerable<WeatherForecast> Get()
+    {
+        this.logger.LogInformation("WeatherForecast GET called");
+        WeatherForecast[] result;
+        using (var activity = ActivitySource.StartActivity("Getting weather data.", ActivityKind.Internal))
         {
-            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            result = GetWeatherData();
+            activity?.SetTag("Got data.", result);
```

Comment:
is result one of the primitive types? if not exporter's will likely won't understand it.